code/__HELPERS/unsorted.dm

```diff
@@ -1708,4 +1708,12 @@ var/mob/dview/dview_mob = new
 
 	see_in_dark = 1e6
 
-/mob/dview/New() //For whatever reason, if this isn't called, then BYOND will throw a type mismatch runtime when attempting to add this to the mobs list. -Fox
\ No newline at end of file
+/mob/dview/New() //For whatever reason, if this isn't called, then BYOND will throw a type mismatch runtime when attempting to add this to the mobs list. -Fox
+
+/proc/IsValidSrc(var/A)
+	if(istype(A, /datum))
+		var/datum/B = A
+		return !isnull(B.gcDestroyed)

```

Comment:
`gcDestroyed` is null for anything that hasn't been qdel'd, which means you probably want to take the `!` that's up here...